nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/proto/ProtoUtils.java

```diff
@@ -46,8 +48,22 @@ public static DynamicMessage createMessage(Descriptors.Descriptor descriptor, Ma
            switch (field.getType()) {
            case MESSAGE:
                if (field.isRepeated()) {
-                   Collection collection = value.getClass().isArray() ? Arrays.asList((Object[]) value) : (Collection) value;
-                   collection.forEach(act -> builder.addRepeatedField(field, createMessage(field.getMessageType(), (Map<String, Object>) act, tableSchema)));
+                   final Collection<Map<String, Object>> valueMaps;
+                   if (value instanceof Object[] arrayValue) {
+                       valueMaps = Arrays.stream(arrayValue)
+                               .map(item -> (Map<String, Object>) item).toList();
+                   } else if (value instanceof HashMap<?, ?> hashMapValue) {
```

Comment:
If value is a Map, is it always guaranteed to be specifically a `HashMap`?

Reply:
Good catch @dan-s1 . 
I also wanted to address that but didn't find the time to answer until now. 

I don't think we should assume that. Also there's no benefit of doing so, as we only ever use methods of `Map<?, ?>` anyways. 

```suggestion
                   } else if (value instanceof Map<?, ?> mapValue) {
```